### ISSUE TITLE
Add JSR-330 support module

### ISSUE THREAD

**Issue by jbee** [jbee/silk#73]
Use reflection to check if one of its classes is on the CP (`Class.forName`). If so the required annotation classes can be resolved from class names (`String`) using reflection and used to configure the bootstrapping to have the effect specified by JSR-330.

**Comment by Michael1993** [comment]
Wouldn't that go against the design philosophy of purejin?

I'm a bit confused, JSR-330 is a set of annotations (e.g.: `@Inject`) and the intro/readme document says purejin explicitly does **not** depend on annotations.

**Comment by jbee** [comment]
@Michael1993 yes, it would. This is strictly an idea to make transitions more appealing. Reduce the amount of work needed for a codebase to switch to an annotation-less way. It is part of purejins philosophy to allow gradual transition in any direction really. 
What I don't see happening is full CDI support or something similar as CDI and purejin are fundamentally different and any attempt to bridge this would either fail or be huge pain and pile of complexity we just try to get away from.
